### ISSUE TITLE
Error: (0x80041010) Invalid class Source: SWbemServicesEx Specifically: ExecNotificationQueryAsync

### ISSUE THREAD

**Issue by vb0** [XMCQCX/DeviceIDPnP#4]
I'm getting the above in : 037: WMI.ExecNotificationQueryAsync(Sink, "SELECT * FROM __InstanceCreationEvent " . command)
from DeviceIDFinder.ahk.

Everything was working fine, but on one of the latest W10 updates I noticed DeviceIDPnP behaves like nothing plugged in (though no errors). I decided to run DeviceIDFinder.ahk which gives this error and if ignored shows no devices (so I guess the root cause it the same for both).

I'm sure I've been using AHK2 since forever for this, but to make sure I downloaded the latest version and still does the same.

**Comment by XMCQCX** [comment]
After running it on both Windows 10 and Windows 11, I didn’t experience any errors. I haven’t come across this error before, but here’s my recommendation: These commands should be safe to run. However, for added safety, consider creating a backup image of your PC. Perform each step and check if it resolves the issue after each one.
- Check for Windows 10 updates.
- Restart the PC.
- In an elevated command prompt, run the following commands:
`DISM /Online /Cleanup-Image /RestoreHealth`
`sfc /scannow`

- Restarts WMI service. run this code in an AHK script (v2)
`RunWait('*RunAs Powershell.exe -Command "Restart-Service -Name winmgmt -Force"',, 'Hide')`

**Comment by vb0** [comment]
Thanks, I'm kind of weary of these generic DISM/sfc commands so I went searching for the specific error and how troubleshot it, and [got here](https://techcommunity.microsoft.com/t5/ask-the-performance-team/wmi-missing-or-failing-wmi-providers-or-invalid-wmi-class/ba-p/375485). Then (even if probably I could follow the workflow and narrow down the problem) I got lazy again and took the shotgun approach to rebuild everything as per one of the comments there (pasted below too for reference). And then things started working again, for some reason DeviceIDFinder.ahk got rid of the error but didn't show anything, might be my mistake for not running it as Admin, in any case for good measure run the thing the second time and rebooted, everything is fine now! Bonus: Veeam Agent's service (which I couldn't bring up since about the same time) started working again! Thanks for putting me on the right track, I was sure MS borked something for everyone!

```
*** Recompile Mof files for WMI errors ***
-run from elevated cmd prompt:
cd C:\Windows\System32\Wbem
for /f %s in ('dir /b *.mof *.mfl') do mofcomp %s
for %i in (*.dll) do regSvr32 -s %i
net stop winmgmt /y
net start winmgmt
gpupdate /force
```
